In [0]:
from google.colab import drive
drive.mount('/content/drive')

In [0]:
import pandas as pd
import numpy as np

df_train = pd.read_csv("/content/drive/My Drive/Colab Notebooks/Kaggle/20191102_Titnic/train.csv")
df_test = pd.read_csv("/content/drive/My Drive/Colab Notebooks/Kaggle/20191102_Titnic/test.csv")

df_all = pd.concat([df_train, df_test], axis=0, sort=False)
# 不要そうな項目削除
df_all = df_all.drop(['PassengerId', 'Name', 'Ticket'], axis=1)

# 欠損値補完
df_all = df_all.fillna(df_all.median())
# カテゴリ変数のone-hot化
df_all = pd.get_dummies(df_all)

# 訓練データ
train = df_all.iloc[:len(df_train)] 
# テストデータ
test = df_all.iloc[len(df_train):] 
test = df_all.drop('Survived', axis=1)

# 訓練データとラベルの分割
X = train.drop('Survived', axis=1)
y = train.Survived

In [0]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)

# ロジスティック回帰
from sklearn.linear_model import LogisticRegression
logistic = LogisticRegression()
logistic.fit(X_train, y_train)
print(logistic.score(X_test, y_test))

# 決定木
from sklearn.tree import DecisionTreeClassifier
tree = DecisionTreeClassifier()
tree.fit(X_train, y_train)
print(tree.score(X_test, y_test))

0.7982062780269058
0.8026905829596412


/usr/local/lib/python3.6/dist-packages/sklearn/linear_model/logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


0.6278026905829597 # 'SibSp', 'Parch'

0.7892376681614349 # 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Survived'

0.7982062780269058 # 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Survived', 'Cabin', 'Embarked'


In [0]:
len(df_test)

418

In [0]:
y

0      0.0
1      1.0
2      1.0
3      1.0
4      0.0
      ... 
886    0.0
887    1.0
888    0.0
889    1.0
890    0.0
Name: Survived, Length: 891, dtype: float64

In [0]:
aaa = pd.read_csv("/content/drive/My Drive/Colab Notebooks/Kaggle/20191102_Titnic/train.csv")
bbb = aaa.fillna(df_all.median())
aaa['PassengerId'] = aaa['PassengerId'].fillna(method='pad')

In [0]:
df_train

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0              1         0       3  ...   7.2500   NaN         S
1              2         1       1  ...  71.2833   C85         C
2              3         1       3  ...   7.9250   NaN         S
3              4         1       1  ...  53.1000  C123         S
4              5         0       3  ...   8.0500   NaN         S
..           ...       ...     ...  ...      ...   ...       ...
886          887         0       2  ...  13.0000   NaN         S
887          888         1       1  ...  30.0000   B42         S
888          889         0       3  ...  23.4500   NaN         S
889          890         1       1  ...  30.0000  C148         C
890          891         0       3  ...   7.7500   NaN         Q

[891 rows x 12 columns]